### ISSUE TITLE
Vim9: cannot use imported function as completion function for user-defined Ex command

### ISSUE THREAD

**Issue by lacygoill** [vim/vim#9541]
**Steps to reproduce**

Run this shell command:

    vim -Nu NONE -S <(tee <<'EOF'
        vim9script
        var dir = '/tmp/import'
        dir->mkdir('p')
        var lines =<< trim END
            vim9script
            export def Complete(..._): list<string>
                return ['abcd']
            enddef
        END
        lines->writefile(dir .. '/script.vim')
        import '/tmp/import/script.vim'
        command -nargs=1 -complete=customlist,script.Complete Cmd Func(<q-args>)
        def Func(arg: string)
            echomsg arg
        enddef
        feedkeys(":Cmd ab\<Tab>", 'nt')
    EOF
    )

`E117` is given:

    E117: Unknown function: script.Complete

**Expected behavior**

No error is given.  The argument after `:Cmd` is completed from `ab` into `abcd`.

**Version of Vim**

8.2 Included patches: 1-4103

**Environment**

Operating system: Ubuntu 20.04.3 LTS
Terminal: xterm
Value of $TERM: xterm-256color
Shell: zsh 5.8

**Additional context**

Saving a funcref of the imported function inside a variable, and using the latter to refer to the completion function does not help:

    var Ref: func = script.Complete
    command -nargs=1 -complete=customlist,Ref Cmd Func(<q-args>)

Test:
```vim
vim9script
var dir = '/tmp/import'
dir->mkdir('p')
var lines =<< trim END
    vim9script
    export def Complete(..._): list<string>
        return ['abcd']
    enddef
END
lines->writefile(dir .. '/script.vim')
import '/tmp/import/script.vim'
var Ref: func = script.Complete
command -nargs=1 -complete=customlist,Ref Cmd Func(<q-args>)
def Func(arg: string)
    echomsg arg
enddef
feedkeys(":Cmd ab\<Tab>", 'nt')
```
    E117: Unknown function: Ref

Prefixing the funcref with `s:` (aka script-local namespace) does not help either:

    var Ref: func = script.Complete
    command -nargs=1 -complete=customlist,s:Ref Cmd Func(<q-args>)
                                          ^^

Test:
```vim
vim9script
var dir = '/tmp/import'
dir->mkdir('p')
var lines =<< trim END
    vim9script
    export def Complete(..._): list<string>
        return ['abcd']
    enddef
END
lines->writefile(dir .. '/script.vim')
import '/tmp/import/script.vim'
var Ref: func = script.Complete
command -nargs=1 -complete=customlist,s:Ref Cmd Func(<q-args>)
def Func(arg: string)
    echomsg arg
enddef
feedkeys(":Cmd ab\<Tab>", 'nt')
```
    E117: Unknown function: s:Ref

---

No issue when `Complete()` is not imported, but defined in the same script:
```vim
vim9script
def Complete(..._): list<string>
    return ['abcd']
enddef
command -nargs=1 -complete=customlist,Complete Cmd Func(<q-args>)
def Func(arg: string)
    echomsg arg
enddef
feedkeys(":Cmd ab\<Tab>", 'nt')
```
    :Cmd abcd

---

As a workaround, we need to `:execute` the `:command` so that `script.Complete` is evaluated into a funcref.

    execute printf('command -nargs=1 -complete=customlist,%s Cmd Func(<q-args>)', script.Complete)

Test:
```vim
vim9script
var dir = '/tmp/import'
dir->mkdir('p')
var lines =<< trim END
    vim9script
    export def Complete(..._): list<string>
        return ['abcd']
    enddef
END
lines->writefile(dir .. '/script.vim')
import '/tmp/import/script.vim'
execute printf('command -nargs=1 -complete=customlist,%s Cmd Func(<q-args>)', script.Complete)
def Func(arg: string)
    echomsg arg
enddef
feedkeys(":Cmd ab\<Tab>", 'nt')
```
    :Cmd abcd

But this is cumbersome, and looks awkward.

---

There was no issue with the previous import mechanism:
```vim
vim9script
var dir = '/tmp/import'
dir->mkdir('p')
var lines =<< trim END
    vim9script
    export def Complete(..._): list<string>
        return ['abcd']
    enddef
END
lines->writefile(dir .. '/script.vim')
import Complete from '/tmp/import/script.vim'
command -nargs=1 -complete=customlist,Complete Cmd Func(<q-args>)
def Func(arg: string)
    echomsg arg
enddef
feedkeys(":Cmd ab\<Tab>", 'nt')
```
    :Cmd abcd

So I guess it's a regression introduced in [8.2.4019](https://github.com/vim/vim/releases/tag/v8.2.4019).

**Comment by monkoose** [comment]
@lacygoill is this the reason why your vim9-syntax currently doesn't work?

**Comment by lacygoill** [comment]
The github version must be broken since the last change of the import mechanism.  I've pushed an update to fix it, but I'm not sure how reliable it is.  I need to refactor the import.  But there are still quite a few issues to report.  [A week ago](https://github.com/vim/vim/releases/tag/v8.2.4050), the first commit introducing the new autoload mechanism said that it was only the first step:

   > Solution:   First step in supporting "vim9script autoload" and "import
   >             autoload".

I tried to start the refactoring, but I immediately noticed #9496, as well as other issues; so, I was waiting for the feature to be fully implemented before going further.
